java/cli/src/main/java/org/ray/cli/RayCli.java

```diff
@@ -0,0 +1,320 @@
+    // Register app to Redis. 
+    byte[] zip = FileUtil.fileToBytes(cmdSubmit.packageZip);
+
+    String packageName = cmdSubmit.packageZip.substring(
+      cmdSubmit.packageZip.lastIndexOf('/') + 1,
+      cmdSubmit.packageZip.lastIndexOf('.'));
+
+    //final RemoteFunctionManager functionManager = RayRuntime
+    //    .getInstance().getRemoteFunctionManager();
+
+    UniqueID resourceId = functionManager.registerResource(zip);
+    RayLog.rapp.debug(
+        "registerResource " + resourceId + " for package " + packageName + " done");
+
+    UniqueID appId = params.driver_id;
+    functionManager.registerApp(appId, resourceId);
+    RayLog.rapp.debug("registerApp " + appId + " for resouorce " + resourceId + " done");
+  
+    // Unzip the package file.
+    String appDir = "/tmp/" + cmdSubmit.className;
+    String extPath = appDir + "/" + packageName;
+    if (!FileUtil.createDir(extPath, false)) {
+      throw new RuntimeException("create dir " + extPath + " failed ");
+    }
+
+    ZipFile zipFile = new ZipFile(cmdSubmit.packageZip);
+    zipFile.extractAll(extPath);
+
+    // Build the args for driver process.
+    File originDirFile = new File(extPath);
+    File[] topFiles = originDirFile.listFiles();
+    String topDir = null;
+    for (File file : topFiles) {
+      if (file.isDirectory()) {
+        topDir = file.getName();
+      }
+    }
+    RayLog.rapp.debug("topDir of app classes: "+ topDir);
+    if (topDir == null) {
+      RayLog.rapp.error("Can't find topDir of app classes, the app directory " + appDir);
+      return;
+    }
+
+    String additionalClassPath = appDir + "/" + packageName  + "/" + topDir + "/*";
+    RayLog.rapp.debug("Find app class path  " + additionalClassPath);
+
+    // Start driver process.
+    //RunManager runManager = new RunManager(params, RayRuntime.getInstance().getPaths(),
+    //  RayRuntime.configReader);
+    RunManager runManager = new RunManager(params, paths, config);
+    Process proc = runManager.startDriver(
+      DefaultDriver.class.getName(),
+      cmdSubmit.redis_address,
+      appId,
+      appDir,
+      params.node_ip_address,
+      cmdSubmit.className,
+      cmdSubmit.classArgs,
+      additionalClassPath,
+      null);
+
+    if (null == proc) { 
+      RayLog.rapp.error(
+        "Create process for app " + packageName + " in local directory " + appDir
+            + " failed");
+      return;
+    }
+
+    RayLog.rapp
+    .info("Create app " + appDir + " for package " + packageName + " succeeded");
+  }
+
+  private static String getConfigPath(String config) throws Exception {
+    String configPath;
+
+    if (config != null && !config.equals("")) {
+      configPath = config;
+    } else {
+      configPath = System.getenv("RAY_CONFIG");
+      if (configPath == null) {
+        configPath = System.getProperty("ray.config");
+      }
+      if (configPath == null) {
+        throw new Exception("Please set config file path in env RAY_CONFIG or property ray.config");
```

Comment:
better use a more specific exception, `RuntimeException` or `IllegalArgumentException`.
Then this function doesn't have to be declared with `throws exception`. So is its caller.

Reply:
thanks.